### PR TITLE
improvement(core): log aborted nodes on dep error

### PR DESCRIPTION
**Pull request by thsig**
**What this PR does / why we need it**:

When a dependency fails, we now log a line for each aborted node.

Also fixed the counting logic used in the command error summary. Before this fix, we used to include aborted nodes in the failed count.

**Special notes for your reviewer**:

Note that we're no longer copying the error from the failed dependency to all the failed dependants in the command result. This might be seen as a breaking change.

If we'd rather keep the machine-readable command output the same, I can modify the logic to set the error for the aborted dependant keys in the command result.

### CODE REVIEW

File: core/src/graph/nodes.ts [hunk 1]
```diff
@@ -146,14 +146,32 @@ export abstract class TaskNode<T extends Task = Task> {
...
+        if (depKey === task.getKey() && d.executionType === "request" && error) {
+          depAborted = false
+          depError = new GraphNodeError({ resultError: error, node: d })
+        } else {
+          depAborted = true
+          depError = null
+          if (!keys.has(depKey)) {
+            d.task.log.info({
+              msg: `Aborting because upstream dependency failed.`,
+              metadata: metadataForLog(d.task, "error", inputVersion),
```
Comment (eysi09) [inline core/src/graph/nodes.ts:164]: Not part of this PR per se but we should really try and do away with this metadata field. 

I don't think it's really used anywhere and we already have the `context` field which serves the same purpose.

File: core/src/commands/base.ts [hunk 2]
```diff
@@ -1053,8 +1054,14 @@ export async function handleProcessResults(
       return f && f.error ? [toGardenError(f.error)] : []
     })
 
+    const errMsg = abortedCount
+      ? failedCount
+        ? `${failedCount} requested ${taskType} action(s) failed, ${abortedCount} aborted!`
+        : `${abortedCount} requested ${taskType} action(s) aborted!`
+      : `${failedCount} requested ${taskType} action(s) failed!`
```
Comment (stefreak) [inline core/src/commands/base.ts:1061]: This is hard to review due to the cyclomatic complexity of this function; Do we have tests for this error message?

Looking at tests it would be much easier to tell what the error message looks like now.

**Review by stefreak — APPROVED** [review]
The code and the added test looks good to me, but the updated status counting code for the error message is hard to understand. Adding a test for it might be a good idea (optional).
